source/setup.Rmd

```diff
@@ -1,282 +1,277 @@
+
+### MacOS
+
+**Installation** To install Docker on MacOS,
+visit [the online Docker documentation](https://docs.docker.com/desktop/install/mac-install/), and
+download the `Docker.dmg` installation file that is appropriate for your
+computer. To know which installer is right for your machine, you need to know 
+whether your computer has an Intel processor (older machines) or an
+Apple processor (newer machines); the [Apple support page](https://support.apple.com/en-ca/HT211814) has
+information to help you determine which processor you have. Once downloaded, double-click 
+the file to open the installer, then drag the Docker icon to the Applications folder.
+Double-click the icon in the Applications folder to start Docker. In the installation 
+window, use the recommended settings.
+
+**Running JupyterLab** Once Docker is installed and running, follow the
+instructions below in the Windows section on *Running JupyterLab* (the user
+interface is the same). More information on installing and running Docker on
+MacOS, as well as troubleshooting tips, can be 
+found [in the online Docker documentation](https://docs.docker.com/desktop/install/mac-install/).
+
+### Windows
+
+**Installation** To install Docker on Windows, visit [the online Docker documentation](https://docs.docker.com/desktop/install/windows-install/),
+and download the `Docker Desktop Installer.exe` file. Double-click the file to open the installer
+and follow the instructions on the installation wizard, choosing **WSL-2** instead of **Hyper-V** when prompted.
+
+Occasionally, when you first run Docker on Windows, you will encounter an error message. Some common errors you may see:
+
+- If you need to update WSL, you can enter `cmd.exe` in the Start menu to run the command line. Type `wsl --update` to update WSL.
+- If the admin account on your computer is different to your user account, you must add the user to the "docker-users" group. 
+  Run Computer Management as an administrator and navigate to `Local Users` and `Groups -> Groups -> docker-users`. Right-click to 
+  add the user to the group. Log out and log back in for the changes to take effect.
+- If you need to enable virtualization, you will need to edit your BIOS. Restart your computer, and enter the BIOS using the hotkey
+  (usually Delete, Esc, and/or one of the F# keys). Look for an "Advanced" menu, and under your CPU settings, set the "Virtualization" option
+  to "enabled". Then save the changes and reboot your machine. If you are not familiar with BIOS editing, you may want to find an expert 
+  to help you with this, as editing the BIOS can be dangerous. Detailed instructions for doing this are beyond the scope of this book. 
+More information on installing and running
+Docker on Windows, as well as troubleshooting tips, can be found [in the online
+Docker documentation](https://docs.docker.com/desktop/install/windows-install/).
```

Comment:
I think this should be presented as a note, as it's only needed if something goes wrong.

Reply:
Good idea, I will change that